snowpack/src/commands/build.ts

```diff
@@ -343,9 +343,10 @@ class FileBuilder {
 
   async getProxy(originalFileLoc: string) {
     const proxiedCode = this.output[originalFileLoc];
-    const proxiedUrl = originalFileLoc
-      .substr(this.config.buildOptions.out.length)
-      .replace(/\\/g, '/');
+    const proxiedUrl = (
+      this.config.buildOptions.baseUrl +
+      originalFileLoc.substr(this.config.buildOptions.out.length).replace(/\\/g, '/')
+    ).replace(/((?:[^:]|^)\/)\/+/g, '$1');
```

Comment:
Could this be simplified by using URL?

```js
const proxyUrlPath = originalFileLoc.substr(this.config.buildOptions.out.length).replace(/\\/g, '/');
const proxiedUrl = new URL(this.config.buildOptions.baseUrl, proxyUrlPath).href
```

Reply:
Good point, although couldn't that be solved by changing the URL path to be relative to the baseUrl?

```js
console.log(new URL('./d/e/f', 'http://example.com/a/b/c').href))

// http://example.com/a/b/d/e/f
```

Reply:
Ah, didn't know you could use a dot to make a URL relative like that! Agreed, that's much nicer - the only issue I see if that the user needs to make sure their base URL ends in a trailing `/`, although with how it's currently used, I think it already needs to, so this shouldn't be too big of a deal. I'll get that change in :smiley:

Reply:
Oh, there's actually an issue here - when `buildOptions.baseUrl` is unset, it defaults to `/`, which the `URL` constructor complains isn't a valid URL.